aten/src/ATen/TensorImpl.cpp

```diff
@@ -41,6 +42,23 @@ void Tensor::backward(
   pImpl->backward(std::move(gradient), keep_graph, create_graph);
 }
 
+TensorImpl::TensorImpl(Backend backend, ScalarType scalar_type) {
+  type_ = &globalContext().getType(backend, scalar_type);
```

Comment:
There shouldn't be any references to `globalContext()` from `TensorImpl`. So (not in this patch, necessarily) eventually we should just take the storage directly as an argument to the constructor.